### PR TITLE
Sort versions as numbers in lifecycle display

### PR DESCRIPTION
**Pull request by mjgiarlo**
It is surprising to scroll to the bottom of an object with 30+ versions and see the last entry, which one would reasonably assume to be the latest version, show up as version 9.

### CODE REVIEW

**Comment by jcoyne** [comment]
@mjgiarlo in all the noise there was:
```
#<SyntaxError: /home/travis/build/sul-dlss/argo/app/views/catalog/_show_milestones.html.erb:42: syntax error, unexpected end-of-input, expecting end>
```

File: app/views/catalog/_show_milestones.html.erb [hunk 1]
```diff
@@ -4,27 +4,25 @@
...
+        steps = milestones[version]
+        first = true
         steps.each_pair do |k,m| -%>
-        <% if first
-            first=false
-        -%>
-          <tr style="display:none" class="last_step<%= version%>">
-            <td><span style="cursor: pointer;" onclick='$(".version<%= version%>").toggle();$(".last_step<%= version%>").toggle();'">+</span><%= version%></td>
+          <% if first
+               first = false
```
[the remark this answers is not in the document]
Comment (jcoyne) [reply app/views/catalog/_show_milestones.html.erb]: missing `end` tag?

[the remark this answers is not in the document]
Comment (jcoyne) [reply app/views/catalog/_show_milestones.html.erb]: missing `end` tag?

[the remark this answers is not in the document]
Comment (mjgiarlo) [reply app/views/catalog/_show_milestones.html.erb]: I believe that's on line 29.

**Comment by mjgiarlo** [comment]
> Won’t this only draw the first row?

I don't *think* my PR affects that determination. Here's the diff between what's in my branch here and what the code looked like before I busted stuff in #1812:

```diff
diff --git a/app/views/catalog/_show_milestones.html.erb b/app/views/catalog/_show_milestones.html.erb                                                
index f2f1c1ec..be81e3a4 100644                                                                                                                       
--- a/app/views/catalog/_show_milestones.html.erb                                                                                                     
+++ b/app/views/catalog/_show_milestones.html.erb                                                                                                     
@@ -4,27 +4,25 @@
   </thead>
   <tbody>
     <%
-      ## TODO: Pick a real date format for the date to_s calls.
-      ## Do it here and NOT in the model!!
       version_hash = document.get_versions
       milestones = document.milestones
-      milestones.keys.sort.each do |version|
-        steps=milestones[version]
-        first=true
+      milestones.keys.sort_by { |version_number_string| version_number_string.to_i }.each do |version|
+        steps = milestones[version]
+        first = true
         steps.each_pair do |k,m| -%>
-        <% if first
-            first=false
-        -%>
-          <tr style="display:none" class="last_step<%= version%>">
-            <td><span style="cursor: pointer;" onclick='$(".version<%= version%>").toggle();$(".last_step<%= version%>").toggle();'">+</span><%= vers
ion%></td>
+          <% if first
+               first = false
+          -%>
+          <tr style="display:none" class="last_step<%= version %>">
+            <td><span style="cursor: pointer;" onclick='$(".version<%= version %>").toggle();$(".last_step<%= version %>").toggle();'>+</span><%= ver
sion %></td>
             <td><%= steps['accessioned'][:display] || k.titleize %></td>
-            <td><%= steps['accessioned'][:time].nil? ? 'pending' : steps['accessioned'][:time].in_time_zone.to_s() %></td>
+            <td><%= steps['accessioned'][:time].nil? ? 'pending' : steps['accessioned'][:time].in_time_zone.to_s %></td>
           </tr>
-          <tr class="version<%= version%>">
-            <td colspan=3 onclick='$(".version<%= version%>") .toggle();$(".last_step<%= version%>").toggle();'><span style="cursor: pointer;">-</spa
n>
-              <%=version%>
+          <tr class="version<%= version %>">
+            <td colspan=3 onclick='$(".version<%= version %>").toggle();$(".last_step<%= version %>").toggle();'><span style="cursor: pointer;">-</sp
an>
+              <%= version %>
               <% if version_hash[version] -%>
-                (<%=version_hash[version][:tag]%>) <%=version_hash[version][:desc]%>
+                (<%= version_hash[version][:tag] %>) <%= version_hash[version][:desc] %>
               <% end -%>
             </td>
           </tr>
@@ -32,7 +30,7 @@
         <tr class="version<%= version %>">
           <td></td>
           <td><%= m[:display] || k.titleize %></td>
-          <td><%= m[:time].nil? ? 'pending' : m[:time].in_time_zone.to_s() %></td>
+          <td><%= m[:time].nil? ? 'pending' : m[:time].in_time_zone.to_s %></td>
         </tr>
       <% end -%>
     <% end -%>
```

**Pull request merged**